pass-deposit-services/deposit-core/src/main/java/org/eclipse/pass/deposit/support/deploymenttest/DspaceDepositService.java

```diff
@@ -0,0 +1,220 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestClient;
+
+/**
+ * @author Russ Poetker (rpoetke1@jh.edu)
+ */
+@Service
+public class DspaceDepositService {
+    private static final Logger LOG = LoggerFactory.getLogger(DspaceDepositService.class);
+
+    private final RestClient restClient;
+
+    @Value("${dspace.user}")
+    private String dspaceUsername;
+
+    @Value("${dspace.password}")
+    private String dspacePassword;
+
+    @Value("${dspace.server}")
+    private String dspaceServer;
+
+    @Value("${dspace.server.api.protocol}")
+    private String dspaceApiProtocol;
+
+    protected record AuthContext(String xsrfToken, String authToken){}
+
+    public DspaceDepositService(@Value("${dspace.server.api.protocol}") String dspaceApiProtocol,
+                                @Value("${dspace.server}") String dspaceServer,
+                                @Value("${dspace.server.api.path}") String dspaceApiPath) {
+        PoolingHttpClientConnectionManager connectionManager = PoolingHttpClientConnectionManagerBuilder.create()
+            .setDefaultSocketConfig(SocketConfig.custom()
+                .setSoTimeout(Timeout.ofMinutes(1))
+                .build())
+            .setDefaultConnectionConfig(ConnectionConfig.custom()
+                .setSocketTimeout(Timeout.ofMinutes(1))
+                .setConnectTimeout(Timeout.ofMinutes(1))
+                .setTimeToLive(TimeValue.ofMinutes(10))
+                .build())
+            .build();
+        final CloseableHttpClient httpClient = HttpClients.custom()
+            .setConnectionManager(connectionManager)
+            .disableCookieManagement()
+            .build();
+        this.restClient = RestClient.builder()
+            .requestFactory(new HttpComponentsClientHttpRequestFactory(httpClient))
+            .baseUrl(dspaceApiProtocol + "://" + dspaceServer + dspaceApiPath)
+            .build();
+    }
+
+    /**
+     * Authenticate with the repository. This should be called as appropriate per the repository's API
+     * authentication docs.
+     * @return an AuthContext containing authToken and xsrfToken
+     */
+    AuthContext authenticate() {
+        // Using exchange is needed for this call because dspace returns 404, but the response headers has the
+        // csrf token header DSPACE-XSRF-TOKEN
+        ResponseEntity<Void> csrfResponse = restClient.get()
+            .uri("/security/csrf")
+            .exchange((request, response) -> new ResponseEntity<>(null, response.getHeaders(), HttpStatus.OK));
+        String xsrfToken = csrfResponse.getHeaders().get("DSPACE-XSRF-TOKEN").get(0);
```

Comment:
May want to null check on this in case `DSPACE-XSRF-TOKEN` is null or empty list. Possibly log if it's null. This would be helpful with debugging.

Reply:
Done